Composer/packages/lib/indexers/src/utils/diagnosticUtil.ts

```diff
@@ -44,6 +46,22 @@ export function filterTemplateDiagnostics(diagnostics: Diagnostic[], template: L
   });
 }
 
+export function filterSectionDiagnostics(diagnostics: Diagnostic[], section: LuIntentSection): Diagnostic[] {
+  const { range } = section;
+  if (!range) return diagnostics;
+  const filteredDiags = diagnostics.filter(d => {
+    return d.range && d.range.start.line >= range.startLineNumber && d.range.end.line <= range.endLineNumber;
```

Comment:
I've seen this code before, it's duplicated with some of my previous comments.
Avoid repeating it.